nav2_util/include/nav2_util/simple_action_server.hpp

```diff
@@ -74,6 +84,41 @@ class SimpleActionServer
       handle_accepted);
   }
 
+  void activate()
+  {
+    std::lock_guard<std::mutex> lock(activation_mutex_);
+    server_active_ = true;
+  }
+
+  void deactivate()
+  {
+    std::lock_guard<std::mutex> lock_server_state(activation_mutex_);
+    std::lock_guard<std::mutex> lock_goal_handle(update_mutex_);
+
+    server_active_ = false;
+
+    // TODO(orduno) Replace with `abort_all()` once #849 is merged
+    if (current_handle_ != nullptr && current_handle_->is_active()) {
+      RCLCPP_WARN(node_->get_logger(), "Taking action server to deactive state "
+        " with an active goal. Cancelling the goal.");
+      current_handle_->abort(std::make_shared<typename ActionT::Result>());
+      current_handle_.reset();
+    }
+
+    if (new_handle_ != nullptr && new_handle_->is_active()) {
+      RCLCPP_WARN(node_->get_logger(), "Taking action server to deactive state "
+        " with an active goal. Cancelling the goal.");
```

Comment:
Should these warning differentiate between the current and new handle?

Reply:
Will change this.